### ISSUE TITLE
Lint/Debugger triggered by false positive

### ISSUE THREAD

**Issue by ilvez** [rubocop/rubocop#11572]
## Expected behavior

Rubocop should consider `{ P: p }` correct code. In this case `p` is used as name of variable/method and isn't related to debugging call.

## Actual behavior

```
% echo "{ P: p }" | rubocop --stdin foobar --only Lint/Debugger --debug
For /home/ilvez: Default configuration from /home/ilvez/.asdf/installs/ruby/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.45.1/config/default.yml
Inspecting 1 file
Scanning /home/ilvez/foobar
W

Offenses:

foobar:1:6: W: Lint/Debugger: Remove debugger entry point p.
{ P: p }
     ^

1 file inspected, 1 offense detected
Finished in 0.09155614499468356 seconds
```

## RuboCop version

```
% rubocop -V
1.45.1 (using Parser 3.2.1.0, rubocop-ast 1.26.0, running on ruby 3.1.2) [x86_64-linux]
```

**Comment by bbatsov** [comment]
That's fixed on master. See #11564.

**Comment by ilvez** [comment]
Thanks, next time I'll check against master branch as well.
